lib/internal/modules/helpers.js

```diff
@@ -63,6 +63,16 @@ function toRealPath(requestPath) {
   });
 }
 
+/** @type {Set<string>} */
+let userConditions;
+
+function getUserConditions() {
+  if (userConditions === undefined) {
+    userConditions = new SafeSet(getOptionValue('--conditions'));
```

Comment:
https://github.com/nodejs/node/blob/9abd1c7288d173ac697d9798a39372fbe4f28594/doc/contributing/primordials.md#L429-L445

```suggestion
    userConditions = new SafeSet(new SafeArrayIterator(getOptionValue('--conditions')));
```